src/TriggerActivityMakerLowEnergyEvent.cpp

```diff
@@ -0,0 +1,269 @@
+    if (config.contains("trigger_on_adjacency"))
+      m_trigger_on_adjacency = config["trigger_on_adjacency"];
+    if (config.contains("adj_tolerance"))
+      m_adj_tolerance = config["adj_tolerance"];
+    if (config.contains("adjacency_threshold"))
+      m_adjacency_threshold = config["adjacency_threshold"];
+  }
+
+}
+
+TriggerActivity
+TriggerActivityMakerLowEnergyEvent::construct_ta(Window m_current_window) const
+{
+
+  TriggerPrimitive latest_tp_in_window = m_current_window.inputs.back();
+
+  TriggerActivity ta;
+  ta.time_start = m_current_window.time_start;
+  ta.time_end = latest_tp_in_window.time_start + latest_tp_in_window.time_over_threshold;
+  ta.time_peak = latest_tp_in_window.time_peak;
+  ta.time_activity = latest_tp_in_window.time_peak;
+  ta.channel_start = latest_tp_in_window.channel;
+  ta.channel_end = latest_tp_in_window.channel;
+  ta.channel_peak = latest_tp_in_window.channel;
+  ta.adc_integral = m_current_window.adc_integral;
+  ta.adc_peak = latest_tp_in_window.adc_peak;
+  ta.detid = latest_tp_in_window.detid;
+  ta.type = TriggerActivity::Type::kTPC;
+  ta.algorithm = TriggerActivity::Algorithm::kLowEnergyEvent;
+  ta.inputs = m_current_window.inputs;
+
+  return ta;
+}
+
+uint16_t
+TriggerActivityMakerLowEnergyEvent::check_adjacency(Window window) const
+{
+  // This function returns the adjacency value for the current window, where adjacency
+  // is defined as the maximum number of consecutive wires containing hits. It accepts
+  // a configurable tolerance paramter, which allows up to adj_tolerance missing hits
+  // on adjacent wires before restarting the adjacency count. The maximum gap is 4 which
+  // comes from tuning on December 2021 coldbox data, and June 2022 coldbox runs.
+
+  uint16_t adj = 1;              // Initialise adjacency, 1 for the first wire.
+  uint16_t max = 0;              // Maximum adjacency of window, which this function returns
+  unsigned int channel = 0;      // Current channel ID
+  unsigned int next_channel = 0; // Next channel ID
+  unsigned int next = 0;         // The next position in the hit channels vector
+  unsigned int tol_count = 0;    // Tolerance count, should not pass adj_tolerance
+
+  // Generate a channelID ordered list of hit channels for this window
+  std::vector<int> chanList;
+  for (auto tp : window.inputs) {
+    chanList.push_back(tp.channel);
+  }
+  std::sort(chanList.begin(), chanList.end());
+
+  // ADAJACENCY LOGIC ====================================================================
+  // =====================================================================================
+  // Adjcancency Tolerance = Number of times prepared to skip missed hits before resetting 
+  // the adjacency count. This accounts for things like dead channels / missed TPs. The 
+  // maximum gap is 4 which comes from tuning on December 2021 coldbox data, and June 2022 
+  // coldbox runs.
+  for (int i = 0; i < chanList.size(); ++i) {
```

Comment:
Use `size_t i` to avoid compiler complaints

Reply:
Done